### DISCUSSION TITLE
Does loader get revalidated every time an action happens?

### DISCUSSION THREAD

**Discussion by juniorforlife** [remix-run/remix#7417] [Q&A] [answered]
I’m building a web app like Tinder. Each time a card is swiped, an action is called. User can swipe really fast. Does this mean the page loader will refetch data every single time user swipes? Will this cause any performance issue or timeout error?

**Comment by sergiodxa** [comment, marked as answer]
Yes it will be revalidated after every action, it can cause perf issues if your loader is slow, you can use the shouldRevalidate function to avoid the loader to run again if you don't need more data after the user swipe.
